netscaler/netscaler_resource.go

```diff
@@ -96,25 +115,45 @@ func readResponseHandler(resp *http.Response) ([]byte, error) {
 	}
 }
 
-func (c *NitroClient) createHTTPRequest(method string, url string, buff *bytes.Buffer) (*http.Request, error) {
-	req, err := http.NewRequest(method, url, buff)
+func (c *NitroClient) createHTTPRequest(method string, urlstr string, buff *bytes.Buffer) (*http.Request, error) {
+	req, err := http.NewRequest(method, urlstr, buff)
 	if err != nil {
 		return nil, err
 	}
+	// Get resourceType from url
+	u, err := url.Parse(urlstr)
+	if err != nil {
+		return nil, err
+	}
+	splitStrings := strings.Split(u.Path, "/")
+	resourceType := splitStrings[len(splitStrings)-1]
+
 	req.Header.Set("Accept", "application/json")
 	req.Header.Set("Content-Type", "application/json")
 	if c.proxiedNs == "" {
-		req.Header.Set("X-NITRO-USER", c.username)
-		req.Header.Set("X-NITRO-PASS", c.password)
+		if len(c.sessionid) > 0 {
+			req.Header.Set("Set-Cookie", "NITRO_AUTH_TOKEN="+c.sessionid)
+		} else {
+			if resourceType != "login" {
+				req.Header.Set("X-NITRO-USER", c.username)
+				req.Header.Set("X-NITRO-PASS", c.password)
+			}
+		}
 	} else {
```

Comment:
does session-based work with proxied NS as well?

Reply:
ADM as an API proxy does have support for session but expects it in 'SESSID' header. Removed the code in case of proxied NS as this is outside the scope of this PR. 


Reference: https://docs.citrix.com/en-us/citrix-application-delivery-management-software/13/adm-as-api-proxy-server.html